### PR TITLE
TVOS-151: Sample project

### PR DESCRIPTION
**Pull request by huebnerob**
# Ticket(s)

https://vimean.atlassian.net/browse/TVOS-151

# Ticket Summary

Clean up and improve the sample project.  Provide a taste of how to do things with `VimeoNetworking`.

# Implementation Summary

- On launch, app tries to load an account, and falls back to client credentials if it didn’t/couldn’t load one.
- If client credentials gets a 40X response, we prompt the user to double check their client identifier and secret, which is the most likely culprit.
- In MasterViewController, we listen for account changed notifications and update the table view of video names to suit.  Logged in shows /me/videos and logged out shows /channels/staffpicks/videos.
- Log In/Log Out button in nav bar launches code grant auth URL if logged out or logs out if logged in.

# How to test

Get a clientID/secret from the api site and plug them in.  Make sure to set up the redirect URL things too (see readme).

### CODE REVIEW

File: VimeoNetworking/VimeoNetworking/AuthenticationController.swift [hunk 1]
```diff
@@ -307,7 +307,14 @@ final public class AuthenticationController
             }
         }
         
-        self.client.authenticatedAccount = nil
+        if loadClientCredentials
+        {
+            self.client.authenticatedAccount = (try? self.accountStore.loadAccount(.ClientCredentials)) ?? nil
```
Comment (alfiehanssen) [inline VimeoNetworking/VimeoNetworking/AuthenticationController.swift]: If `try` fails should we clear those values from account store?

Comment (alfiehanssen) [inline VimeoNetworking/VimeoNetworking/AuthenticationController.swift]: The bogus values I mean, assuming failure from corrupted data

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift [hunk 2]
```diff
@@ -7,42 +7,25 @@
...
-    var client: VimeoClient?
-    
-    var observationToken: ObservationToken?
 
     func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool
     {
         // Override point for customization after application launch.
         let splitViewController = self.window!.rootViewController as! UISplitViewController
         splitViewController.delegate = self
         let navigationController = splitViewController.viewControllers[splitViewController.viewControllers.count-1] as! UINavigationController
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift]: another possibility? `splitViewController.viewControllers.last`

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift [hunk 3]
```diff
@@ -55,108 +38,59 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
...
-    {
-        // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
-    }
-
-    func applicationWillTerminate(application: UIApplication)
-    {
-        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
-    }
-    
     // MARK: - URLs
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift]: 👍

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift [hunk 4]
```diff
@@ -8,95 +8,172 @@
...
+//            {
+//                let object = videos[indexPath.row] as! NSDate
+//                let controller = (segue.destinationViewController as! UINavigationController).topViewController as! DetailViewController
+//                controller.detailItem = object
+//                controller.navigationItem.leftBarButtonItem = self.splitViewController?.displayModeButtonItem()
+//                controller.navigationItem.leftItemsSupplementBackButton = true
+//            }
+//        }
+//    }
+
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift]: "uncomment to enable..."

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift [hunk 5]
```diff
@@ -8,95 +8,172 @@
...
     // MARK: - Table View
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift]: UITableViewDataSource protocol implementation, or Table view data source protocol

**Comment by nicolelehrer** [comment]
This might be intentional but there was a url scheme (vimeo{Client_ID} saved in the project I pulled, fyi

**Comment by nicolelehrer** [comment]
Tested with clientID/secret from the api site and plugged them in, plus redirect URL, on simulator, works great 👍

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift [hunk 6]
```diff
@@ -7,42 +7,25 @@
...
-            // Fallback on earlier versions
-        }
+        navigationController.topViewController!.navigationItem.leftBarButtonItem = splitViewController.displayModeButtonItem()
         
-        let client = VimeoClient(appConfiguration: self.appConfiguration)
-        self.client = client
         
-        let authenticationController = AuthenticationController(client: client)
-        self.authenticationController = authenticationController
+        let authenticationController = AuthenticationController(client: VimeoClient.defaultClient)
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift]: nit space above this

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift [hunk 7]
```diff
@@ -55,108 +38,59 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
...
-    }
-
-    func applicationWillTerminate(application: UIApplication)
-    {
-        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
-    }
-    
     // MARK: - URLs
     
+
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift]: nit - space

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift [hunk 8]
```diff
@@ -55,108 +38,59 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
...
-        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
-    }
-    
     // MARK: - URLs
     
+    
     func application(app: UIApplication, openURL url: NSURL, options: [String : AnyObject]) -> Bool
     {
-        self.authenticationController?.codeGrant(responseURL: url) { result in
+
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift]: nit - space

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift [hunk 9]
```diff
@@ -8,95 +8,172 @@
 
 import UIKit
 
-class MasterViewController: UITableViewController {
-
+class MasterViewController: UITableViewController
+{
     var detailViewController: DetailViewController? = nil
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift]: need to set to nil?

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift [hunk 10]
```diff
@@ -8,95 +8,172 @@
...
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
-        self.navigationItem.leftBarButtonItem = self.editButtonItem()
-
-        let addButton = UIBarButtonItem(barButtonSystemItem: .Add, target: self, action: #selector(insertNewObject))
-        self.navigationItem.rightBarButtonItem = addButton
-        if let split = self.splitViewController {
+        
+        if let split = self.splitViewController
+        {
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift]: seems here we if-let unwrap but below force in `self.splitViewController!.collapsed`

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift [hunk 11]
```diff
@@ -8,95 +8,172 @@
...
-
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
+    
+    // MARK: - Setup
+    
+    private func setupAccountObservation()
+    {
+
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift]: space

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift [hunk 12]
```diff
@@ -8,95 +8,172 @@
...
-                } else {
-                    // Fallback on earlier versions
-                }
-                controller.navigationItem.leftItemsSupplementBackButton = true
+    
+    // MARK: - Actions
+    
+    @objc private func didTapAuthenticationButton()
+    {
+
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift]: space

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift [hunk 13]
```diff
@@ -8,95 +8,172 @@
...
+        let authenticationController = AuthenticationController(client: VimeoClient.defaultClient)
+        if VimeoClient.defaultClient.isAuthenticatedWithUser
+        {
+            do
+            {
+                try authenticationController.logOut()
             }
+            catch let error as NSError
+            {
+
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift]: space

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift [hunk 14]
```diff
@@ -8,95 +8,172 @@
...
 
+    // MARK: - Segues
+
+//    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?)
+//    {
+//        if segue.identifier == "showDetail"
+//        {
+//            if let indexPath = self.tableView.indexPathForSelectedRow
+//            {
+//                let object = videos[indexPath.row] as! NSDate
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift]: cast as NSDate?

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/DetailViewController.swift [hunk 15]
```diff
@@ -35,23 +32,6 @@ class DetailViewController: UIViewController {
         super.viewDidLoad()
         // Do any additional setup after loading the view, typically from a nib.
         self.configureView()
-
```
Comment (nicolelehrer) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/DetailViewController.swift]: how about displaying a feature of VIMVideo e.g. `label.text = detail.videoDescription` in `func configureView()`

**Comment by nicolelehrer** [comment]
@huebnerob 🔥 🔥 🔥 DWR - Tested with clientID/secret from the api site and plugged them in, plus redirect URL on simulator, works great

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift (hunk 2 above)
Comment (huebnerob) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/AppDelegate.swift]: this is actually Apple's default code 😆 

but I'll clean it up for them

File: VimeoNetworking/VimeoNetworking/AuthenticationController.swift (hunk 1 above)
Comment (huebnerob) [inline VimeoNetworking/VimeoNetworking/AuthenticationController.swift]: Yessir, I'll add this functionality directly to `AccountStore`, for anytime a fetch errors

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift (hunk 10 above)
Comment (huebnerob) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift]: Apple again 😱

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift (hunk 14 above)
Comment (huebnerob) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOS/MasterViewController.swift]: 👍

**Comment by huebnerob** [comment]
Thanks @nicolelehrer!

**Pull request merged**